src/main/resources/com/puppycrawl/tools/checkstyle/grammar/java.g

```diff
@@ -346,14 +346,14 @@ typeArguments[boolean addImagNode]
         lt:LT {#lt.setType(GENERIC_START); ;ltCounter++;}
         // (Dinesh Bolkensteyn) Added support for Java 7 diamond notation
         // (disabled ambiguous warnings since generated code seems to work)
-        (options{generateAmbigWarnings=false;}:typeArgument[addImagNode]
+        (options{generateAmbigWarnings=false;}:typeArgument[false]
```

Comment:
it looks like argument `addImagNode` is not needed on line 342.
Same for typeArgument node (line 326)

Reply:
I have remove the argument from `typeArguments` but the argument for `typeArgument` is still required for the calls to the other "type" productions, i.e. `classTypeSpec`, `builtInTypeSpec`, and `wildcardType`.

Reply:
the only invocations of typeArgument are in this node and its argument is always false. So argument can be removed and usages inside typeArgument changed to false.

Reply:
Good point. Done.